### PR TITLE
[PE-253] Add Github Actions

### PR DESCRIPTION
**Pull request by Synicalx**
## Description, Motivation and Context

- Migrating this repo away from CircleCI
- Step 1 here is adding the Github Actions config
- Note: this is currently set to run on all PR's so it can be tested here prior to merging, this will need to be changed in a subsequent PR

## Checklist:

- [x] I've added/updated tests to cover my changes
- [x] I've created an issue associated with this PR

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    runs-on: runs-on,runner=4cpu-linux-x64
+    steps:
+    - uses: actions/checkout@v4
+    - uses: ./.github/actions/prepare-repository
+
+    - name: Run lint checker
+      run: yarn lint:check
+
+  publish:
+    runs-on: runs-on,runner=4cpu-linux-x64
```
Comment (vlj91) [inline .github/workflows/build-and-test.yml:70]:
```suggestion
    runs-on: runs-on,runner=4cpu-linux-x64
    if: github.event_name == 'release'
```

**suggestion:** This job should only run on the `publish` event type

**Review by vlj91 — DISMISSED** [review]
LGTM apart from the conditional on the `publish` job

**Review by timdawborn — COMMENTED** [review]
LGTM apart from the fact that this PR is missing dependabot configurations for both of the actions and the workflows.

**Review by vlj91 — APPROVED** [review]

**Review by timdawborn — APPROVED** [review]
LGTM 👍

**Pull request merged**